docs/specifications/linking-and-embedding-specification.md

```diff
@@ -0,0 +1,364 @@
+
+### EBlocks
+
+While anchors define a single location in the document, eblocks define two; the beginning and the end of an eblock. The intent of a block is to specify content bounded by the begin and end markers. The end of an eblock can either be explicit or implicit.
+
+**Explicit eblocks** are defined with the anchor syntax and have two special rules; 1) the name of the beginning anchor starts with `begin:` and the ending anchor starts with `end:`, and 2) the other parts of the name must match. An example of matching anchors is:
+ 
+  ```markdown
+  <a name="begin:my-block"></a>
+  Sunt cillum occaecat adipisicing aliquip excepteur
+  nisi ea nostrud do nostrud.
+
+  Ad magna labore tempor dolor dolore nulla magna Lorem
+  magna qui et. Ipsum eiusmod minim ad est nostrud sunt occaecat.
+  <a name="end:my-block"></a>
+  ```
+
+**Implicit eblocks** are defined by any anchor whose name is not starting with `begin:` or `end:`, and the point where the nesting level of the first non-immediate AST node either remains on the same level or decreases in comparison to the starting place of the anchor. The nesting level is changed e.g. with different levels of headings and bullet list indents. Additionally, the eblock can also be started with a heading, in which case the eblock ends when the chapter following the heading ends. In case the end of a file is found before the nesting level of the markdown content decreases, eblock ends there. The example below illustrate implicit eblocks with various eblock definitions.
+
+```markdown
+# My Note
+
+Occaecat esse minim Lorem fugiat mollit eu eiusmod magna dolor.
+Velit ea cupidatat pariatur amet pariatur est est laboris culpa
+ipsum magna eu nulla esse.
+
+## Second Level
+
+<a name="y"></a>
+Cupidatat velit esse est anim consequat enim ad.
+
+- Nostrud incididunt
+- <a name="x"></a> consequat non dolore
+  - consequat fugiat
+  - eiusmod esse ipsum
+    - anim nulla aute
+- ad reprehenderit
+
+## Another Second Level
+
+Ut reprehenderit sint quis pariatur. Non ullamco eiusmod nulla
+do sunt pariatur magna fugiat. Ullamco nulla magna nostrud
+in aute eu laborum ex est nostrud nostrud excepteur Lorem.
+```
+
+Examples of implicit eblocks defined above:
+
+- `my-note` refers to the whole note (id of the H1 heading of the note).
+- `second-level` refers to the "Second Level" chapter (stopping before the heading "Another Second Level")
+- `y` refers to the text `Cupidatat velit esse est anim consequat enim ad.`
+- `x` refers to the following list fragment:
+  ```markdown
+  - consequat non dolore
+    - consequat fugiat
+    - eiusmod esse ipsum
+      - anim nulla aute
+- `another-second-level` refers to the "Another Second Level" chapter (stopping at the end of the file)
+
+### Aliases
+
+Aliases define alternative names for notes. They are defined in an optional front matter, using an attribute "aliases". The example below shows how this can be done using YAML front matter:
+
+```YAML
+---
+aliases:
```

Comment:
Not sure if this is mentioned elsewhere, but I'd also like to consider adding a primary "title: Title" attribute-value pair that can override the markdown heading of the page. Sometimes for presentational reasons it's inconvenient to name a document in a way that makes for a good title elsewhere, or to have a H1 tag at all.